e2etest/webhookstests/webhooks.go

```diff
@@ -106,6 +113,51 @@ var _ = ginkgo.Describe("Webhooks", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring("overlaps with already defined CIDR"))
 		})
+
+		ginkgo.DescribeTable("IPAddressPool with overlapping addresses of the nodes",
+			func(ipFamily ipfamily.Family) {
+				nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				nodeIps, err := k8s.NodeIPsForFamily(nodes.Items, ipFamily, "")
+				Expect(err).NotTo(HaveOccurred())
+				for _, nodeIPString := range nodeIps {
+					nodeIP := net.ParseIP(nodeIPString)
+					var nodeCIDRs []string
+					if nodeIP.To4() != nil {
+						cidr := &net.IPNet{
+							IP:   nodeIP,
+							Mask: net.CIDRMask(32, 32),
+						}
+						nodeCIDRs = append(nodeCIDRs, cidr.String())
+					} else {
+						cidr := &net.IPNet{
+							IP:   nodeIP,
+							Mask: net.CIDRMask(128, 128),
+						}
+						nodeCIDRs = append(nodeCIDRs, cidr.String())
+					}
+					ginkgo.By("Creating IPAddressPool")
+					resources := config.Resources{
+						Pools: []metallbv1beta1.IPAddressPool{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "webhooks-test3",
+								},
+								Spec: metallbv1beta1.IPAddressPoolSpec{
+									Addresses: nodeCIDRs,
+								},
+							},
+						},
+					}
+					err = ConfigUpdater.Update(resources)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("contains nodeIp"))
+					break
```

Comment:
here I break out the loop after the first node.

Reply:
but then why building the nodCIDRs? It's always going to be one right?

Reply:
or, even doing the loo in first instance, if we always break

Reply:
agree. I will change.